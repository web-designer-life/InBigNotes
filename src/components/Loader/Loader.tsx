import React, { Component } from 'react';
import {
    LoaderBackground,
    LoaderContainer,
    LoaderText,
} from './style';
import loader from '../../images/loader.gif';
import { TEXTS } from '../../constants';

export default class Loader extends Component {
    constructor(props) {
        super(props);

        this.state = {};
    }

    render() {
        return (
            <LoaderBackground>
                <LoaderContainer>
                    <img
                        src={loader}
                        width={75}
                        height={75}
                        alt={TEXTS.LOADER}
                    />
                    <LoaderText>{TEXTS.LOADER}</LoaderText>
                </LoaderContainer>
            </LoaderBackground>
        )
    }
};
