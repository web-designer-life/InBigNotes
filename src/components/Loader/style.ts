import styled from 'styled-components';

const Background = styled.div`
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    background-color: #000000;
    opacity: .7;
    z-index: 100;
`;

const LoaderContent = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 125px;
    height: 125px;
    color: #FFFFFF;
    margin: auto;
`;

const LoaderText = styled.p`
    color: #FFFFFF;
`;

export { Background, LoaderContent, LoaderText };