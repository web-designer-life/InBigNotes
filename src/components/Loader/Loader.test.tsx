import React from "react";
import Loader from "./Loader";
import renderer from 'react-test-renderer';

describe("Loader component", () => {
    it("should render Loader component", () => {
        const component = renderer.create(<Loader />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
