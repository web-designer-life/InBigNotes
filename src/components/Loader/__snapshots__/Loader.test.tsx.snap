// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render Loader component 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="Loading..."
      height={75}
      src={null}
      width={75}
    />
    <styled.p>
      Loading...
    </styled.p>
  </styled.div>
</styled.div>
`;
